### PR TITLE
Changes to support multiple local_networks and code refactor

### PR DESCRIPTION
**Pull request by muresan**
### Summary and Scope

- Fixes:
- Requires:
- Relates to:

#### Issue Type

- Bugfix Pull Request
- Docs Pull Request
- RFE Pull Request

### Prerequisites

- [ ] I have included documentation in my PR (or it is not required)
- [ ] I tested this on internal system (if yes, please include results or a description of the test)
- [ ] I tested this on a vshasta system (if yes, please include results or a description of the test)
 
### Risks and Mitigations
 

# NOTES

this branch requires that you have a local `management` network created before deployment. If you don't need it, comment it from inventory/defaults.yaml

### CODE REVIEW

File: modules/hypervisor/variables.tf [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  default = "/"
+}
+
+variable "luks_keys" {
+  type    = map(string)
+  default = {}
+}
+
+variable "hypervisor_name" {
+  type = string
```
Comment (manderson-hpe) [inline modules/hypervisor/variables.tf:51]: Should this have a default? Do we explicitly not want a default?

File: modules/inventory/main.tf [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+locals {
+  inventory = merge([for f in fileset("${var.data_path}", "inventory/*.{yaml,json}") : yamldecode(file(format("${var.data_path}/%s", f)))]...)
```
Comment (manderson-hpe) [inline modules/inventory/main.tf:26]: will a json file work with yamldecode? Docs suggest jsondecode. Might just remove the json extension and force yaml.

**Review by manderson-hpe — APPROVED** [review]

File: modules/hypervisor/variables.tf (hunk 1 above)
Comment (muresan) [reply modules/hypervisor/variables.tf:51]: No, this needs to come as a parameter when we call the module.

File: modules/inventory/main.tf (hunk 2 above)
Comment (muresan) [reply modules/inventory/main.tf:26]: Yes,. JSON is a subset of YAML and I've tested, yamldecode loads both json and yaml. Which makes this much much simpler.

File: prod/inventory/inventory.yaml [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
+  prefix: kubernetes
 
 hypervisor_defaults: {}
 
+network_defaults:
+  addresses: []
+  dhcp4: false
+  dhcp6: false
+  static: false
+  mtu: 1500
```
Comment (rustydb) [inline prod/inventory/inventory.yaml]:
```suggestion
  mtu: 9000
```

File: prod/inventory/inventory.yaml [hunk 4]
```diff
@@ -1,8 +1,19 @@
 globals:
   env_name: prod
+  prefix: kubernetes
 
 hypervisor_defaults: {}
 
+network_defaults:
+  addresses: []
+  dhcp4: false
```
Comment (rustydb) [inline prod/inventory/inventory.yaml]:
```suggestion
  dhcp4: true
```

File: modules/hypervisor/main.tf [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  environment         = var.environment
+  prefix              = var.prefix
+  vcpu                = each.value.vcpu
+  memory              = each.value.memory
+  name                = each.key
+  hypervisor_name     = var.hypervisor_name
+  network_interfaces  = each.value.network_interfaces
+  pci_devices         = each.value.pci_devices
+  base_volume         = each.value.base_volume
+  # this grabs the network_id for the ones we created
```
Comment (rustydb) [inline modules/hypervisor/main.tf]:
```suggestion
```

File: modules/hypervisor/main.tf [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  local_networks      = [ for k,v in each.value.local_networks : merge(
+    v,
+    (v.create ? { "id" = module.network.id[v.name]} : {})
+  )]
+  roles               = each.value.roles
+  volumes             = each.value.volumes
+  ssh_keys            = each.value.ssh_keys
+}
+
+# change to for_each and use network
```
Comment (rustydb) [inline modules/hypervisor/main.tf:47]: is this a todo?

File: modules/inventory/outputs.tf [hunk 7]
```diff
@@ -0,0 +1,11 @@
+output "hypervisors" {
```
Comment (rustydb) [inline modules/inventory/outputs.tf:25]: Add copyright header.

File: modules/inventory/variables.tf
Comment (rustydb) [inline modules/inventory/variables.tf:1]: https://github.com/Cray-HPE/fawkes-terraform/pull/14/files#r1370550466

File: scripts/get-pci-devices-ids.sh
Comment (rustydb) [inline scripts/get-pci-devices-ids.sh:1]: Missing copyright.

File: dev/inventory/defaults.yaml [hunk 8]
```diff
@@ -10,6 +10,13 @@ hypervisor_defaults:
     static: false
     mtu: 1500
     mode: none
+  local_networks:
+    - name: isolated
+      addresses:
+        - "192.168.201.0/24"
+        #- "2001:db8:ca2:2::1/64"
+    - name: management
+      create: false
```
Comment (rustydb) [inline dev/inventory/defaults.yaml]:
```suggestion
#    - name: management
#      create: false
```

File: modules/hypervisor/main.tf (hunk 6 above)
Comment (muresan) [reply modules/hypervisor/main.tf:47]: not really, it's just to look like the above module instance, it's fine either way. For now let's leave it like this.

File: scripts/get-pci-devices-ids.sh [hunk 9]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
```
Comment (rustydb) [inline scripts/get-pci-devices-ids.sh]:
```suggestion
```

File: scripts/prod-proxy.sh [hunk 10]
```diff
@@ -1,5 +1,30 @@
 #!/usr/bin/env bash
 
+#
```
Comment (rustydb) [inline scripts/prod-proxy.sh]:
```suggestion
#
```

**Review by rustydb — APPROVED** [review]

**Pull request merged**